Source/Core/Core/IOS/USB/Emulated/Skylanders/SkylanderCrypto.cpp

```diff
@@ -0,0 +1,145 @@
+      0x9969, 0xA90A, 0xB92B, 0x5AF5, 0x4AD4, 0x7AB7, 0x6A96, 0x1A71, 0x0A50, 0x3A33, 0x2A12,
+      0xDBFD, 0xCBDC, 0xFBBF, 0xEB9E, 0x9B79, 0x8B58, 0xBB3B, 0xAB1A, 0x6CA6, 0x7C87, 0x4CE4,
+      0x5CC5, 0x2C22, 0x3C03, 0x0C60, 0x1C41, 0xEDAE, 0xFD8F, 0xCDEC, 0xDDCD, 0xAD2A, 0xBD0B,
+      0x8D68, 0x9D49, 0x7E97, 0x6EB6, 0x5ED5, 0x4EF4, 0x3E13, 0x2E32, 0x1E51, 0x0E70, 0xFF9F,
+      0xEFBE, 0xDFDD, 0xCFFC, 0xBF1B, 0xAF3A, 0x9F59, 0x8F78, 0x9188, 0x81A9, 0xB1CA, 0xA1EB,
+      0xD10C, 0xC12D, 0xF14E, 0xE16F, 0x1080, 0x00A1, 0x30C2, 0x20E3, 0x5004, 0x4025, 0x7046,
+      0x6067, 0x83B9, 0x9398, 0xA3FB, 0xB3DA, 0xC33D, 0xD31C, 0xE37F, 0xF35E, 0x02B1, 0x1290,
+      0x22F3, 0x32D2, 0x4235, 0x5214, 0x6277, 0x7256, 0xB5EA, 0xA5CB, 0x95A8, 0x8589, 0xF56E,
+      0xE54F, 0xD52C, 0xC50D, 0x34E2, 0x24C3, 0x14A0, 0x0481, 0x7466, 0x6447, 0x5424, 0x4405,
+      0xA7DB, 0xB7FA, 0x8799, 0x97B8, 0xE75F, 0xF77E, 0xC71D, 0xD73C, 0x26D3, 0x36F2, 0x0691,
+      0x16B0, 0x6657, 0x7676, 0x4615, 0x5634, 0xD94C, 0xC96D, 0xF90E, 0xE92F, 0x99C8, 0x89E9,
+      0xB98A, 0xA9AB, 0x5844, 0x4865, 0x7806, 0x6827, 0x18C0, 0x08E1, 0x3882, 0x28A3, 0xCB7D,
+      0xDB5C, 0xEB3F, 0xFB1E, 0x8BF9, 0x9BD8, 0xABBB, 0xBB9A, 0x4A75, 0x5A54, 0x6A37, 0x7A16,
+      0x0AF1, 0x1AD0, 0x2AB3, 0x3A92, 0xFD2E, 0xED0F, 0xDD6C, 0xCD4D, 0xBDAA, 0xAD8B, 0x9DE8,
+      0x8DC9, 0x7C26, 0x6C07, 0x5C64, 0x4C45, 0x3CA2, 0x2C83, 0x1CE0, 0x0CC1, 0xEF1F, 0xFF3E,
+      0xCF5D, 0xDF7C, 0xAF9B, 0xBFBA, 0x8FD9, 0x9FF8, 0x6E17, 0x7E36, 0x4E55, 0x5E74, 0x2E93,
+      0x3EB2, 0x0ED1, 0x1EF0};
+
+  u16 crc = init_value;
+
+  for (u32 i = 0; i < size; i++)
+  {
+    const u16 tmp = (crc >> 8) ^ buffer[i];
+    crc = (crc << 8) ^ CRC_CCITT_TABLE[tmp];
+  }
+
+  return crc;
+}
+// CRC-64 algorithm that is limited to 48 bits every iteration
+u64 SkylanderCrypto::ComputeCRC48(const u8* data, u32 size)
+{
+  const u64 polynomial = 0x42f0e1eba9ea3693;
+  const u64 initialRegisterValue = (static_cast<long long>(2 * 2 * 3) * 1103 * 12868356821);
+
+  u64 crc = initialRegisterValue;
+  for (u32 i = 0; i < size; i++)
+  {
+    crc ^= (static_cast<unsigned long long>(data[i]) << 40);
+    for (char j = 0; j < 8; j++)
+    {
+      if ((crc & 0x800000000000))
+      {
+        crc = (crc << 1) ^ polynomial;
+      }
+      else
+      {
+        crc <<= 1;
+      }
+
+      crc &= 0x0000FFFFFFFFFFFF;
+    }
+  }
+  return crc;
+}
+u64 SkylanderCrypto::CalculateKeyA(u8 sector, const u8* nuid)
+{
+  if (sector == 0)
+  {
+    return (static_cast<u64>(73 * 2017) * 560381651);
+  }
+
+  u8 data[5] = {nuid[0], nuid[1], nuid[2], nuid[3], sector};
+
+  u64 bigEndianCRC = ComputeCRC48(data, 5);
+  u64 littleEndianCRC = 0;
+
+  for (u8 i = 0; i < 6; i++)
+  {
+    u64 byte = (bigEndianCRC >> (i * 8)) & 0xFF;
+    littleEndianCRC += byte << ((5 - i) * 8);
+  }
+
+  return littleEndianCRC;
+}
+void SkylanderCrypto::ComputeChecksum(u8 type, const u8* dataStart, u8* output)
```

Comment:
If possible, try to avoid what we call "magic numbers".  Type should be an [enum class.](https://www.learncpp.com/cpp-tutorial/scoped-enumerations-enum-classes/)